src/PanelTraits/Create.php

```diff
@@ -120,4 +136,86 @@ public function syncPivot($model, $data, $form = 'create')
             }
         }
     }
+
+    private function createOneToOneRelations($item, $data, $form = 'create')
+    {
+        $relationData = $this->formatData($data, $form);
+        $this->createRelationsForItem($item, $relationData);
+    }
+
+    private function createRelationsForItem($item, $formattedData)
+    {
+        foreach ($formattedData['relations'] as $relationMethod => $relationData) {
+            $parent = $item;
+            $model = $relationData['model'];
+            $relation = $parent->{$relationMethod}();
+
+            if ($relation instanceof BelongsTo) {
+                $modelInstance = $model::find($relationData['values'])->first();
+                if ($modelInstance != null) {
+                    $relation->associate($modelInstance)->save();
+                } else {
+                    $relation->dissociate()->save();
+                }
+            } elseif ($relation instanceof HasOne) {
+                if ($parent->{$relationMethod} != null) {
+                    $parent->{$relationMethod}->update($relationData['values']);
+                    $modelInstance = $parent->{$relationMethod};
+                } else {
+                    $relationModel = new $model();
+                    $modelInstance = $relationModel->create($relationData['values']);
+                    $relation->save($modelInstance);
+                }
+            } else {
+                $relationModel = new $model();
+                $modelInstance = $relationModel->create($relationData['values']);
+                $relation->save($modelInstance);
+            }
+
+            if (isset($relationData['relations'])) {
+                $this->createRelationsForItem($modelInstance, ['relations' => $relationData['relations']]);
+            }
+        }
+    }
+
+    private function formatData($data, $form = 'create')
+    {
+        $fieldsWithOneToOneRelations = collect($this->getRelationFields($form))
+            ->sortBy(function ($value) {
+                return substr_count($value['entity'], '.');
+            })
+            ->groupBy('entity')
+            ->filter(function ($value) use ($data) {
+                return array_filter(array_only($data, $value->pluck('name')->toArray()))
+                    && (! isset($value['pivot']) || (0 === strpos($value['type'], 'select')));
+            })
+            ->map(function ($value) use ($data) {
+                $relationArray['model'] = $value->pluck('model')->first();
+                $relationArray['parent'] = $this->getRelationModel($relationArray['model'], -1);
+                $relationArray['values'] = array_only($data, $value->pluck('name')->toArray());
+
+                return $relationArray;
+            })
+            ->all();
```

Comment:
I agree with your sentiment but the whole point of inline comments is to indicate places where it's not obvious why the code choice has been made, an algorithm choice that is not obvious and so forth.

Inline comments are better than valid (for technical or even non-technical reasons) than obscure code.